langchain4j/src/main/java/dev/langchain4j/data/document/loader/ClassPathDocumentLoader.java

```diff
@@ -0,0 +1,353 @@
+	}
+
+	/**
+	 * Loads a {@link Document} from the specified file path.
+	 * <br>
+	 * The file is parsed using the default {@link DocumentParser}.
+	 * The default {@code DocumentParser} is loaded through SPI (see {@link dev.langchain4j.spi.data.document.parser.DocumentParserFactory DocumentParserFactoru}).
+	 * If no {@code DocumentParserFactory} is available in the classpath, a {@link TextDocumentParser} is used.
+	 * <br>
+	 * Returned {@code Document} contains all the textual information from the file.
+	 *
+	 * @param pathOnClasspath The path on the classpath to the file.
+	 * @return document
+	 * @throws IllegalArgumentException If specified path is not a file.
+	 */
+	public static Document loadDocument(String pathOnClasspath) {
+		return loadDocument(pathOnClasspath, DEFAULT_DOCUMENT_PARSER);
+	}
+
+	/**
+	 * Loads a {@link Document} from the specified file path.
+	 * <br>
+	 * The file is parsed using the specified {@link DocumentParser}.
+	 * <br>
+	 * Returned {@code Document} contains all the textual information from the file.
+	 *
+	 * @param pathOnClasspath The path on the classpath to the file.
+	 * @param documentParser The parser to be used for parsing text from the file.
+	 * @return document
+	 * @throws IllegalArgumentException If specified path is not a file.
+	 */
+	public static Document loadDocument(String pathOnClasspath, DocumentParser documentParser) {
+		var classPathSource = ClassPathSource.from(pathOnClasspath);
+
+		try {
+			var uri = classPathSource.getUrl().toURI();
+
+			if (classPathSource.isInsideArchive()) {
+				try (var fs = FileSystems.newFileSystem(uri, Map.of("create", "true"))) {
+					return loadDocument(classPathSource, fs.getPath(pathOnClasspath), documentParser);
+				}
+			}
+			else {
+				return loadDocument(classPathSource, Path.of(uri), documentParser);
+			}
+		}
+		catch (URISyntaxException | IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private static Document loadDocument(ClassPathSource classPathSource, Path path, DocumentParser documentParser) {
+		if (!isRegularFile(path)) {
+			throw illegalArgument("'%s' is not a file", path);
+		}
+
+		return DocumentLoader.load(classPathSource, documentParser);
+	}
+
+	/**
+	 * Loads {@link Document}s from the specified directory. Does not use recursion.
+	 * <br>
+	 * The files are parsed using the default {@link DocumentParser}.
+	 * The default {@code DocumentParser} is loaded through SPI (see {@link dev.langchain4j.spi.data.document.parser.DocumentParserFactory DocumentParserFactoru}).
+	 * If no {@code DocumentParserFactory} is available in the classpath, a {@link TextDocumentParser} is used.
+	 * <br>
+	 * Skips any {@code Document}s that fail to load.
+	 *
+	 * @param folderOnClasspath The path to the directory on the classpath with files.
+	 * @return list of documents
+	 * @throws IllegalArgumentException If specified path is not a directory.
+	 */
+	public static List<Document> loadDocuments(String folderOnClasspath) {
```

Comment:
nit: could you please use "directory" instead of "folder" (everywhere)? AFAIU this is more technically correct name, as well as to keep this consistent with `FileSystemDocumentLoader`